File: RepoTransBench/target_projects/Java/C#/sockeqwe_fragmentargs/public_tests/FragmentArgsPublicTest.cs
using System;
using System.Reflection;
using Xunit;

namespace ProjectName.PublicTests
{
    public class FragmentArgsPublicTest
    {
        public interface IFragmentArgsInjector
        {
            void Inject(object target);
        }

        public class AlternateDummyInjector : IFragmentArgsInjector
        {
            public bool Invoked = false;
            public void Inject(object target)
            {
                if (target != null)
                    Invoked = true;
            }
        }

        public static class FragmentArgs
        {
            public static IFragmentArgsInjector? autoMappingInjector { get; set; }
            public static void Inject(object target)
            {
                autoMappingInjector?.Inject(target);
            }
        }

        [Fact]
        public void TestInjectWithNoAutoMappingClass_DifferentInput()
        {
            FragmentArgs.autoMappingInjector = null;
            FragmentArgs.Inject("publicDummyString");
        }

        [Fact]
        public void TestInjectWithAutoMappingInjectorPresent_DifferentInjector()
        {
            var altInjector = new AlternateDummyInjector();
            FragmentArgs.autoMappingInjector = altInjector;

            int testTarget = 2024;
            FragmentArgs.Inject(testTarget);
            Assert.True(altInjector.Invoked);

            FragmentArgs.autoMappingInjector = null;
        }
    }
}

File: RepoTransBench/target_projects/Java/Go/sockeqwe_fragmentargs/public_tests/fragmentargs_public_test.go
package public_tests

import (
	"testing"
)

type alternateDummyInjector struct {
	invoked bool
}

func (a *alternateDummyInjector) Inject(target interface{}) {
	if target != nil {
		a.invoked = true
	}
}

func TestInjectWithNoAutoMappingClassDifferentInput(t *testing.T) {
	FragmentArgs{}.inject("publicDummyString")
}

func TestInjectWithAutoMappingInjectorPresentDifferentInjector(t *testing.T) {
	altInjector := &alternateDummyInjector{}
	autoMappingInjector = altInjector

	FragmentArgs{}.inject(2024)

	if !altInjector.invoked {
		t.Errorf("Expected injector to be invoked")
	}
	autoMappingInjector = nil
}